Dockerfile

```diff
@@ -0,0 +1,101 @@
+FROM ubuntu:16.04
+
+ENV DEBIAN_FRONTEND=noninteractive
+
+RUN apt-get update && apt-get install --yes --no-install-recommends \
+    automake \
+    bc \
+    bison \
+    ca-certificates \
+    ccache \
+    flex \
+    gawk \
+    gettext \
+    git-core \
+    intltool \
+    liblist-moreutils-perl \
+    liblzo2-dev \
+    libtool \
+    libxml-dom-perl \
+    libxml2-utils \
+    lua5.3 \
+    make \
+    mtd-utils \
+    net-tools \
+    nodejs \
+    pkg-config \
+    qemu-system-arm \
+    qemu-user \
+    rsync \
+    scons \
+    sudo \
+    u-boot-tools \
+    uuid-dev \
+    wget \
+    xutils-dev \
+    xz-utils \
+    zlib1g-dev \
+  && rm --recursive --force /var/lib/apt/lists/* \
+  && ln -sf /usr/bin/lua5.3 /usr/bin/lua \
+  && ln -sf /usr/bin/nodejs /usr/bin/node \
+  && ln -sf /bin/bash /bin/sh
+
+ENV HOME=/home/captain \
+  WSBU_C_COMPILER=/opt/linaro/bin/arm-linux-gnueabihf-gcc \
+  WSBU_CXX_COMPILER=/opt/linaro/bin/arm-linux-gnueabihf-g++ \
+  WSBU_EMULATOR=/usr/bin/qemu-arm \
+  CMAKE_TOOLCHAIN_FILE=/opt/toolchain-linaro-armhf.cmake
+
+COPY toolchain.cmake "$CMAKE_TOOLCHAIN_FILE"
+
+RUN git clone https://github.com/wsbu/linaro-release.git \
+    --branch gcc-linaro-4.9.4-2017.01-x86_64_arm-linux-gnueabihf \
+    --depth 1 \
+    /opt/linaro \
+  && rm -rf /opt/linaro/.git
+
+RUN wget --quiet -O /tmp/cmake.sh https://cmake.org/files/v3.10/cmake-3.10.1-Linux-x86_64.sh \
+  && sh /tmp/cmake.sh --prefix=/usr/local --exclude-subdir --skip-license \
+  && rm /tmp/cmake.sh \
+  && echo "set(CMAKE_C_COMPILER   $WSBU_C_COMPILER)"   >> "$CMAKE_TOOLCHAIN_FILE" \
+  && echo "set(CMAKE_CXX_COMPILER $WSBU_CXX_COMPILER)" >> "$CMAKE_TOOLCHAIN_FILE" \
+  && echo "set(CMAKE_FIND_ROOT_PATH /opt/linaro/arm-linux-gnueabihf)" >> "$CMAKE_TOOLCHAIN_FILE" \
+  && echo "set(CMAKE_CROSSCOMPILING_EMULATOR $WSBU_EMULATOR)" >> "$CMAKE_TOOLCHAIN_FILE"
```

Comment:
Instead of adding these lines in with `>>`, what do you think of simply referencing the environment variables directly in the toolchain file. A line such as this would work:

`set(CMAKE_C_COMPILER "$ENV{WSBU_C_COMPILER}")`

Reply:
I like it. I see you used that pattern elsewhere, just wasn't aware that was a thing. I'll fix it.